src/map_command_handler.hpp

```diff
@@ -190,10 +190,29 @@ class map_command_handler
 		}
 		else if (help_on_unknown_) {
 			utils::string_map symbols;
-			symbols["command"] = get_cmd();
-			symbols["help_command"] = cmd_prefix_ + "help";
-			print("help", VGETTEXT("Unknown command '$command', try $help_command "
-				"for a list of available commands.", symbols));
+			symbols["prefix"] = cmd_prefix_;
+			if (cmd_prefix_ == "/"){
+				symbols["help_command"] = cmd_prefix_ + "help";
+				symbols["command"] = cmd_prefix_ + get_cmd();
+			}
+			else{
+				symbols["help_command"] = "help";
+				symbols["command"] = get_cmd();
+			}
+			if (get_cmd().find(cmd_prefix_) == 0){
+				if (const command* c = get_command(get_data(0).erase(0,1))) {
+					print("help", VGETTEXT("Unknown command '$command', remove '$prefix' or try '$help_command' "
+						"for a list of available commands.", symbols));
+				}
+				else{
+					print("help", VGETTEXT("Unknown command '$command', try '$help_command' "
+						"for a list of available commands.", symbols));
```

Comment:
This and the following `else` print the same message, could merge them by merging the two `if` statements that get to this line.